File: src/SiteKit.php
<?php

namespace ProductShotAI\SiteKit;

/** Public URL helpers for ProductShot AI. */
class SiteKit
{
    public const BASE = 'https://productshotai.app';
    public const BRAND = 'ProductShot AI';
    public const DESCRIPTION = 'AI product photography generator for ecommerce sellers.';

    public static function pageUrl(string $path = '/'): string
    {
        $value = $path === '' ? '/' : $path;
        if ($value[0] !== '/') { $value = '/' . $value; }
        $clean = $value === '/' ? '/' : rtrim($value, '/') . '/';
        return self::BASE . $clean;
    }

    public static function localizedUrl(string $locale, string $path = '/'): string
    {
        if ($locale === 'en') { return self::pageUrl($path); }
        if ($locale === 'zh' || $locale === 'zh-CN') {
            $value = $path === '' ? '/' : $path;
            if ($value[0] !== '/') { $value = '/' . $value; }
            return self::pageUrl('/zh' . ($value === '/' ? '' : $value));
        }
        throw new \InvalidArgumentException('unsupported locale: ' . $locale);
    }

    public static function siteMetadata(): array
    {
        return [
            'name' => self::BRAND,
            'homepage' => self::BASE,
            'description' => self::DESCRIPTION,
            'canonicalPages' => [
                'home' => self::homeUrl(),
                'workbench' => self::workbenchUrl(),
                'pricing' => self::pricingUrl(),
                'blog' => self::blogUrl(),
                'contact' => self::contactUrl(),
                'zhHome' => self::zhHomeUrl(),
            ],
            'tags' => ['productshot', 'ai-product-photography', 'ecommerce-product-photos', 'url-helpers'],
        ];
    }

    public static function homeUrl(): string { return self::pageUrl('/'); }
    public static function workbenchUrl(): string { return self::BASE . '/#workbench'; }
    public static function pricingUrl(): string { return self::BASE . '/#pricing'; }
    public static function blogUrl(): string { return self::pageUrl('/blog'); }
    public static function aboutUrl(): string { return self::pageUrl('/about'); }
    public static function contactUrl(): string { return self::pageUrl('/contact'); }
    public static function privacyUrl(): string { return self::pageUrl('/privacy'); }
    public static function termsUrl(): string { return self::pageUrl('/terms'); }
    public static function refundPolicyUrl(): string { return self::pageUrl('/refund-policy'); }
    public static function zhHomeUrl(): string { return self::localizedUrl('zh', '/'); }
}

File: src/main/java/app/productshotai/sitekit/SiteKit.java
package app.productshotai.sitekit;

import java.util.List;
import java.util.Map;

/** Public URL helpers and metadata for ProductShot AI. */
public final class SiteKit {
    public static final String BASE = "https://productshotai.app";
    public static final String BRAND = "ProductShot AI";
    public static final String DESCRIPTION = "AI product photography generator for ecommerce sellers.";

    private SiteKit() {}

    public static String pageUrl(String path) {
        String value = path == null || path.isEmpty() ? "/" : path;
        if (!value.startsWith("/")) value = "/" + value;
        String clean = value.equals("/") ? "/" : value.replaceAll("/+$", "") + "/";
        return BASE + clean;
    }

    public static String localizedUrl(String locale, String path) {
        if ("en".equals(locale)) return pageUrl(path);
        if ("zh".equals(locale) || "zh-CN".equals(locale)) {
            String value = path == null || path.isEmpty() ? "/" : path;
            if (!value.startsWith("/")) value = "/" + value;
            return pageUrl("/zh" + (value.equals("/") ? "" : value));
        }
        throw new IllegalArgumentException("unsupported locale: " + locale);
    }

    public static Map<String, Object> siteMetadata() {
        return Map.of(
            "name", BRAND,
            "homepage", BASE,
            "description", DESCRIPTION,
            "canonicalPages", Map.of(
                "home", homeUrl(), "workbench", workbenchUrl(), "pricing", pricingUrl(),
                "blog", blogUrl(), "contact", contactUrl(), "zhHome", zhHomeUrl()
            ),
            "tags", List.of("productshot", "ai-product-photography", "ecommerce-product-photos", "url-helpers")
        );
    }

    public static String homeUrl() { return pageUrl("/"); }
    public static String workbenchUrl() { return BASE + "/#workbench"; }
    public static String pricingUrl() { return BASE + "/#pricing"; }
    public static String blogUrl() { return pageUrl("/blog"); }
    public static String aboutUrl() { return pageUrl("/about"); }
    public static String contactUrl() { return pageUrl("/contact"); }
    public static String privacyUrl() { return pageUrl("/privacy"); }
    public static String termsUrl() { return pageUrl("/terms"); }
    public static String refundPolicyUrl() { return pageUrl("/refund-policy"); }
    public static String zhHomeUrl() { return localizedUrl("zh", "/"); }
}
